app/src/main/java/org/wikipedia/views/ImagePreviewDialog.kt

```diff
@@ -0,0 +1,114 @@
+package org.wikipedia.views
+
+import android.content.DialogInterface
+import android.graphics.drawable.Animatable
+import android.net.Uri
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.annotation.NonNull
+import androidx.annotation.Nullable
+import com.facebook.drawee.backends.pipeline.Fresco
+import com.facebook.drawee.controller.BaseControllerListener
+import kotlinx.android.synthetic.main.dialog_image_preview.*
+import org.wikipedia.R
+import org.wikipedia.WikipediaApp
+import org.wikipedia.gallery.ImageInfo
+import org.wikipedia.page.ExtendedBottomSheetDialogFragment
+import org.wikipedia.util.L10nUtil.setConditionalLayoutDirection
+import org.wikipedia.util.StringUtil
+import org.wikipedia.util.UriUtil
+import org.wikipedia.util.log.L
+
+
+class ImagePreviewDialog : ExtendedBottomSheetDialogFragment(), DialogInterface.OnDismissListener {
+
+    private var fileName: String? = null
+    private lateinit var imageInfo: ImageInfo
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        val rootView = inflater.inflate(R.layout.dialog_image_preview, container)
+
+        setConditionalLayoutDirection(rootView, WikipediaApp.getInstance().language().appLanguageCode)
```

Comment:
Just note that the layout direction should rely on the language code from ImageDetails data.

Reply:
I will just make a note for now..and change it with other changes

Reply:
Done